torchvision/datasets/folder.py

```diff
@@ -301,7 +301,7 @@ class ImageFolder(DatasetFolder):
 
     def __init__(
             self,
-            root: str,
+            root: Union[str,bytes,os.PathLike],
```

Comment:
Thanks for the PR @wztdream !

I'm confused, are we really allowing `bytes` type? @pmeier ?

Reply:
As long as we only use it within `os.path` functions, it should be fine. Question is if we want to do that. I'm not sure if it is worth it.

Reply:
yeah, I feel like this isn't something we'd want to explicitly support.
Or at the very least, I'm worried that we'll have to update the type annotation every single time `os.path` decides to support yet another thing